src/sage/crypto/public_key/key_exchange/key_exchange.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
```

Comment:
`from sage.crypto.public_key.key_exchange.key_exchange import KeyExchange` is awkward, but not really avoidable. `from sage.crypto.public_key.key_exchange.key_exchange_base import KeyExchangeBase` is slightly less awkward. Personally I prefer `abstract_key_exchange` but `_base` matches the rest of the Sage library better.

So rename this file and the class. Also change `CommutativeKeyExchange` to `CommutativeKeyExchangeBase`.